lib/AppInfo/Application.php

```diff
@@ -93,10 +94,13 @@ public function register(IRegistrationContext $context): void {
 			$translationService = $container->get(TranslationService::class);
 			$translationService->registerExAppTranslationProviders($context, $container->getServer());
 
-			/** @var TaskProcessingService $taskProcessingService */
-			$taskProcessingService = $container->get(TaskProcessingService::class);
-			$taskProcessingService->registerExAppTaskProcessingProviders($context, $container->getServer());
-			$taskProcessingService->registerExAppTaskProcessingCustomTaskTypes($context);
+			$config = $this->getContainer()->query(IConfig::class);
+			if (version_compare($config->getSystemValueString('version', '0.0.0'), '30.0', '>=')) {
+				/** @var TaskProcessingService $taskProcessingService */
+				$taskProcessingService = $container->get(TaskProcessingService::class);
```

Comment:
I believe auto-import of \OCP\TaskProcessing in the service would throw an error for 28, 29. It would better to use this with an absolute name. Not sure if that needs to be done inside of TaskProcessingService too.

Reply:
this code definitely works on stable28

it calls the `OCA\AppAPI\Service\ProvidersAI\TaskProcessingService` that is inside AppAPI, not the `OCP` one from the Server

Reply:
I thought the imports inside that service file were also imported, so not like python, cool.